imports/wasi_snapshot_preview1/wasmedge.go

```diff
@@ -131,40 +131,54 @@ func (m *Module) WasmEdgeV2SockRecvFrom(ctx context.Context, fd Int32, iovecs Li
 	return Errno(wasi.ESUCCESS)
 }
 
-func (m *Module) WasmEdgeSockSetOpt(ctx context.Context, fd Int32, level Int32, option Int32, value Pointer[Int32], valueLen Int32) Errno {
-	// See socket.go
-	switch wasi.SocketOptionLevel(level) {
-	case wasi.TcpLevel:
-		option += 0x1000
-	}
-	// Only int options are supported for now.
-	switch wasi.SocketOption(option) {
-	case wasi.Linger, wasi.RecvTimeout, wasi.SendTimeout, wasi.BindToDevice:
-		// These accept struct linger / struct timeval / string.
+func (m *Module) WasmEdgeSockSetOpt(ctx context.Context, fd Int32, level Int32, option Int32, value Bytes) Errno {
+	opt := wasi.SocketOption((int64(level) << 32) | int64(option))
+
+	var val wasi.SocketOptionValue
+	switch opt {
+	case wasi.ReuseAddress,
+		wasi.DontRoute,
+		wasi.Broadcast,
+		wasi.SendBufferSize,
+		wasi.RecvBufferSize,
+		wasi.KeepAlive,
+		wasi.OOBInline,
+		wasi.TcpNoDelay:
+
+		if len(value) != 4 {
+			return Errno(wasi.EINVAL)
+		}
+		val = wasi.IntValue(binary.LittleEndian.Uint32(value))
+	case wasi.Linger,
+		wasi.RecvTimeout,
+		wasi.SendTimeout,
+		wasi.BindToDevice:
 		return Errno(wasi.ENOTSUP)
+	case wasi.QuerySocketType,
+		wasi.QuerySocketError,
+		wasi.QueryAcceptConnections:
+		return Errno(wasi.ENOTSUP)
+
+	default:
+		val = wasi.StringValue(value)
```

Comment:
Should we have a `BytesValue` instead? I feel like `StringValue` implies it's UTF-8 but since this is a fallback it could be anything.

Reply:
Given the level of the stack we're operating at, `StringValue` implies that it's a null-terminated string rather than a utf-8 encoded string. `BytesValue` to me reads that it may contain a null byte and so length is supplied separately.

Reply:
At this point of the code we failed to recognized the option and will just pass the bytes as-is to the `wasi.System`, so it may contain null bytes?